typescript/utils/image_utils.ts

```diff
@@ -57,23 +57,23 @@ export async function sourceColorFromImage(image: HTMLImageElement) {
     }
   });
 
-  // Convert Image data to Pixel Array
-  const pixels: number[] = [];
-  for (let i = 0; i < imageBytes.length; i += 4) {
-    const r = imageBytes[i];
-    const g = imageBytes[i + 1];
-    const b = imageBytes[i + 2];
-    const a = imageBytes[i + 3];
-    if (a < 255) {
-      continue;
-    }
-    const argb = argbFromRgb(r, g, b);
-    pixels.push(argb);
+  let ranked: number[];
+
+  if (window.Worker) {
+    const worker = new Worker(new URL('./image_utils_worker.js', import.meta.url), {type: 'module'});
```

Comment:
If I create a new web worker here, do I have to destroy it somehow afterwards? 🤔